framework/include/loops/ComputeLinearFVElementalThread.h

```diff
@@ -0,0 +1,86 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseTypes.h"
+#include "MooseMesh.h"
+#include "MathFVUtils.h"
+
+// libmesh
+#include "libmesh/elem_range.h"
+#include "libmesh/threads.h"
+
+class FEProblemBase;
+class LinearFVElementalKernel;
+
+/**
+ * Adds contributions from volumetric terms discretized using the finite volume method to
+ * the matrix and right hand side of a linear system.
+ */
+class ComputeLinearFVElementalThread
+{
+public:
+  /**
+   * Class constructor.
+   * @param fe_problem Reference to the problem
+   * @param linear_system_num The number of the linear system which is assembled by this thread
+   * @param mode Computation mode (rhs, matrix or both)
+   * @param tags The vector/matrix tags this thread should contribute to. These are used to query
+   * the warehouse for the objects that should contribute to the linear system matrix and right hand
+   * side. When mode == FullSystem, these tags should be vector tags
+   */
+  ComputeLinearFVElementalThread(FEProblemBase & fe_problem,
+                                 const unsigned int linear_system_num,
+                                 const Moose::FV::LinearFVComputationMode mode,
+                                 const std::set<TagID> & tags);
+  /**
+   * Splitting constructor.
+   * @param x Reference to the other thread
+   * @param split The thread split
+   */
+  ComputeLinearFVElementalThread(ComputeLinearFVElementalThread & x, Threads::split split);
+
+  using ElemInfoRange = StoredRange<MooseMesh::const_elem_info_iterator, const ElemInfo *>;
+  /// Operator which is used to execute the thread over a certain iterator range.
+  /// @param range The range of ElemInfos which should be computed.
+  void operator()(const ElemInfoRange & range);
+
+  /// Join threads at the end of the execution
+  void join(const ComputeLinearFVElementalThread & /*y*/);
+
+protected:
+  /// Fetch LinearFVElementalKernels for a given block. We only call this when
+  /// we transition from one block to another.
+  void fetchSystemContributionObjects();
+
+  /// Reference to the problem
+  FEProblemBase & _fe_problem;
+
+  /// The number of the linear system we are contributing to
+  const unsigned int _linear_system_number;
+
+  /// The mode in which this thread is operating
```

Comment:
I would not say this thread. Maybe this loop?
all threads are operating in the same mode